targets/TARGET_Silicon_Labs/TARGET_EFM32/pwmout_api.c

```diff
@@ -334,24 +341,34 @@ void pwmout_period_us(pwmout_t *obj, int us)
     pwmout_period(obj, us / 1000000.0f);
 }
 
+int pwmout_read_period_us(pwmout_t *obj)
+{
+    return TIMER_TopGet(PWM_TIMER) / REFERENCE_FREQUENCY * 1000000;
```

Comment:
You probably want to be more specific about the order of operations here, since in the current form this'll most probably always return 0.